client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -723,18 +744,27 @@ private static WebServiceFeature newMemberSubmissionAddressingFeature() {
     }
 
     public void disconnect() {
-        if (winrm != null && shellSelector != null) {
-            try {
-                winrmCallRetryConnFailure(new CallableFunction<DeleteResponse>() {
-                    @Override
-                    public DeleteResponse call() {
-                        //TODO use different instances of service http://cxf.apache.org/docs/developing-a-consumer.html#DevelopingaConsumer-SettingConnectionPropertieswithContexts
-                        setActionToContext((BindingProvider) winrm, "http://schemas.xmlsoap.org/ws/2004/09/transfer/Delete");
-                        return winrm.delete(null, RESOURCE_URI, MAX_ENVELOPER_SIZE, operationTimeout, locale, shellSelector);
-                    }
-                });
-            } catch (SOAPFaultException soapFault) {
-                assertFaultCode(soapFault, WSMAN_FAULT_CODE_SHELL_WAS_NOT_FOUND);
+        boolean isBusRunning = context.getBus().getState() != BusState.SHUTDOWN;
```

Comment:
Should we guard against `disconnect()` being called twice? If `cleanupContext`, then the second call will throw a `NullPointerException` because `context` will have been set to null.